src/deluge/gui/menu_item/arpeggiator/ratchet_amount.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2014-2023 Synthstrom Audible Limited
+ *
+ * This file is part of The Synthstrom Audible Deluge Firmware.
+ *
+ * The Synthstrom Audible Deluge Firmware is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+#pragma once
+#include "gui/menu_item/unpatched_param.h"
+#include "gui/ui/sound_editor.h"
+
+namespace deluge::gui::menu_item::arpeggiator {
```

Comment:
Instead of 3 separate classes, this, `RatchetProbability`, and `SequenceLength` could be a single class `deluge:gui::menu_item::unpatched_param::Arpeggiator` (in `deluge/gui/menu_item/unpatched_param`).

Reply:
but they are different parameters, how are they supposed to be the same class?

Reply:
The specific parameter to edit is a constructor parameter.

Reply:
I see other params inheriting from  public UnpatchedParam  and they are in different folders and classes. I think I don't understand what you mean

Reply:
All three of these classes are identical (you're just over riding isRelevant) and the param id is passed in the constructor